### ISSUE TITLE
Is explicit selection of desired crate artifacts (libraries/binaries) possible?

### ISSUE THREAD

**Issue by warriorstar-orion** [google/cargo-raze#294]
Hello; as a disclaimer I am not a Rust programmer, I am just trying to fix dependencies somewhere downstack. Sorry in advance if I screw up the terminology or description.

This project: https://github.com/sfackler/rust-phf/tree/master/phf_generator defines a binary, `gen_hash_test`, that appears to be just a test. I'd prefer this not to be included in the list of BUILD targets generated by `cargo raze` (and consequently not vendor any the **binary's** dependencies). My first though thoughts were:

1. Can I provide the specific list of artifacts I want? e.g. just the library itself, not any of the binaries it describes.
2. Can I patch a rust project that I don't explicitly depend on, to remove the binary definition from the `Cargo.toml`?

Are either of these possible? I read through the option code but nothing stood out to me.

Thanks!

**Comment by UebelAndre** [comment]
I don't think I understand the situation here. Cargo raze should only generate things within your dependency graph. In some cases, binaries may be explicitly requested through the use of `extra_aliased_targets`. Can you link the generated build file for `phf_generator`?

**Comment by warriorstar-orion** [comment]
Here you go: https://gist.github.com/warriorstar-orion/224647948cf85dbd6fff1626750e9ccc

Also other pertinent info:

```shell
$ bazel --version
bazel 3.7.0-homebrew

$ uname -a 
Darwin eikthyrnir.local 20.1.0 Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:11 PDT 2020; root:xnu-7195.50.7~2/RELEASE_X86_64 x86_64

$ ack "rules_rust" WORKSPACE -B 1 -A 4                                                                                                                                                   ~/Projects stash~6
http_archive(
    name = "io_bazel_rules_rust",
    sha256 = "b5d4d1c7609714dfef821355f40353c58aa1afb3803401b3442ed2355db9b0c7",
    strip_prefix = "rules_rust-8d2b4eeeff9dce24f5cbb36018f2d60ecd676639",
    urls = [
        # Master branch as of 2020-11-10
        "https://github.com/bazelbuild/rules_rust/archive/8d2b4eeeff9dce24f5cbb36018f2d60ecd676639.tar.gz",
    ],
)

load("@io_bazel_rules_rust//rust:repositories.bzl", "rust_repositories", "rust_repository_set")
load("@io_bazel_rules_rust//proto:repositories.bzl", "rust_proto_repositories")

rust_repositories()

rust_proto_repositories()

load("@io_bazel_rules_rust//:workspace.bzl", "bazel_version", "rust_workspace")

rust_workspace()

bazel_version(name = "bazel_version")

**Comment by UebelAndre** [comment]
I don't believe there is a way to prevent that from being generated but I also don't think it's existence should have a huge impact given the `manual` tag. Can you explain more about the problems you're seeing due to that target being generated?
